### PR TITLE
refactor: component migration

### PR DESCRIPTION
**Pull request by arunjaindev**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Test A
- [ ] Test B

## Checklist

- [ ] The title of the PR states what changed and the related issues number (used for the release note).
- [ ] Does this PR require documentation updates?
- [ ] I've updated documentation as required by this PR.
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: src/Shared/Components/header/index.ts [hunk 1]
```diff
@@ -0,0 +1,2 @@
+export { default as PageHeader } from './PageHeader'
+export * from './HeaderWithCreateButton'
```
Comment (eshankvaish) [inline src/Shared/Components/header/index.ts:2]: export types as well

File: src/Shared/Components/header/HelpNav.tsx [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    ]
+
+    const onClickGettingStarted = (): void => {
+        setGettingStartedClicked(true)
+    }
+
+    const onClickHelpOptions = (option: HelpOptionType): void => {
+        ReactGA.event({
+            category: 'Help Nav',
+            action: `${option.name} Clicked`,
```
Comment (eshankvaish) [inline src/Shared/Components/header/HelpNav.tsx:50]: update casing for header =>Header folder

File: src/Shared/Components/ButtonWIthLoader/ButtonWithLoader.tsx [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    disabled?: boolean
+    rootClassName: string
+    isLoading: boolean
+    onClick?: (event) => void
+    dataTestId?: string
+    children?: React.ReactNode
+    type?: 'submit' | 'reset' | 'button'
+}
+
+export class ButtonWithLoader extends Component<ButtonProps> {
```
Comment (eshankvaish) [inline src/Shared/Components/ButtonWIthLoader/ButtonWithLoader.tsx:15]: We have not exported anywhere, pls verify the usage after exporting for one case

File: src/Common/Constants.ts [hunk 4]
```diff
@@ -286,3 +298,27 @@ export enum ReactSelectInputAction {
...
+
+export const DISCORD_LINK = 'https://discord.devtron.ai/'
+
+export const POSTHOG_EVENT_ONBOARDING = {
+    PREVIEW: 'Preview',
+    DEPLOY_CUSTOM_APP_CI_CD: 'Deploy custom app using CI/CD pipelines',
+    INSTALL_CUSTOM_CI_CD: 'Install CI/CD',
+    VIEW_APPLICATION: 'View helm application',
+    BROWSE_HELM_CHART: 'Browse helm chart',
+    CONNECT_CLUSTER: 'Connect cluster',
```
Comment (eshankvaish) [inline src/Common/Constants.ts:315]: mark deprecated as required

**Review by eshankvaish — CHANGES_REQUESTED** [review]

**Review by vivek-devtron — DISMISSED** [review]

**Review by eshankvaish — APPROVED** [review]

**Review by vivek-devtron — APPROVED** [review]

**Pull request merged**